### PR TITLE
IDE-2062 Import multiple sdk projects base on sdk location in IDE3.0

### PR DESCRIPTION
**Pull request by simonjhy**

### CODE REVIEW

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/simonjhy/liferay-ide/pull/5. [See changes here.](https://github.com/gamerson/liferay-ide/compare/eac98c5~2...eac98c5) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
@simonjhy  also I will make some comments in the code here as well.

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/model/SDKImportProjectsOpMethods.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            public IStatus runInWorkspace( IProgressMonitor monitor ) throws CoreException
+            {
+                final MultiStatusBuilder statusBuilder = new MultiStatusBuilder(ProjectCore.PLUGIN_ID);
+                ElementList<NamedItem>  projectItems = op.getSelectedProjects();
+
+                for( NamedItem namedItem : projectItems )
+                {
+                    try
+                    {
+                        final int startPos = namedItem.getLocation().content().indexOf( "(" );
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/model/SDKImportProjectsOpMethods.java]: This is not a good way to get the project location out of the namedItem.  The namedItem should start the project more directly.  So make that a "projectItem" and store the project location as a variable which will make this easier to handle and you don't have to parse the project name from the string.

File: tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/wizard/ImportSDKProjectsCheckboxCustomPart.java [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.PathBridge;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.RGB;
+
+
+/**
+ * @author Simon Jiang
+ */
+public class ImportSDKProjectsCheckboxCustomPart extends AbstractCheckboxCustomPart
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/wizard/ImportSDKProjectsCheckboxCustomPart.java]: Isn't much of this code duplicated from other classes?  can more be shared here?

Comment (simonjhy) [inline tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/wizard/ImportSDKProjectsCheckboxCustomPart.java]: I have extract common codes to AbstractCheckboxCustomPart class. The most left codes contains very special codes for different target and we can't let them into parene class. The only one we can shared is the SDKImportProjectsLabelProvider class that can be shared with the multiple Binary Projects Import wizard.

Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/wizard/ImportSDKProjectsCheckboxCustomPart.java]: Go ahead and resend your changes.

**Pull request closed without merging**